persist/encoding/decoder.go

```diff
@@ -20,19 +20,28 @@
 
 package encoding
 
-import "github.com/m3db/m3db/persist/schema"
+import (
+	"bytes"
+	"fmt"
+	"io"
+
+	"github.com/m3db/m3db/persist/schema"
+)
 
 // Decoder decodes persisted data
 type Decoder interface {
```

Comment:
Would you mind breaking this into smaller interfaces just to separate the methods a bit. Something like:

```
type FilesetDecoder interface {

}

type CommitlogDecoder interface {
}

type Decoder {
  FilesetDecoder
  CommitlogDecoder
}
```

Reply:
Sure thing.